### PR TITLE
Add namespace and autoloading

### PR DESCRIPTION
**Pull request by cliffordp**
https://central.tri.be/issues/116489

### CODE REVIEW

File: src/Settings.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Tribe\Extensions\Example;
+
+if ( ! class_exists( 'Settings' ) ) {
```
Comment (barryhughes) [inline src/Settings.php]: Needs to be namespace aware.

File: src/Settings.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+			// Remove settings specific to Google Maps
+			add_action( 'admin_init', [ $this, 'remove_settings' ] );
+
+			// Add settings specific to OSM
+			add_action( 'admin_init', [ $this, 'add_settings' ] );
+		}
+
+		/**
+		 * Removes the settings from Events > Settings > General tab > "Map Settings" section
```
Comment (barryhughes) [inline src/Settings.php]: Guessing this is here as an example, but can we make that clear by changing the comment to something like _"Here is an example of removing settings from..."_

File: src/Settings.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	 * Do the Settings.
+	 */
+	class Settings {
+
+		/**
+		 * The Settings Helper class.
+		 *
+		 * @var Settings_Helper
+		 */
+		protected $settings_helper = '';
```
Comment (barryhughes) [inline src/Settings.php]: No need to assign an empty string.

File: src/Settings_Helper.php [hunk 4]
```diff
@@ -0,0 +1,171 @@
+<?php
+
+namespace Tribe\Extensions\Example;
+
+use Tribe__Main;
+
+if ( ! class_exists( 'Settings_Helper' ) ) {
```
Comment (barryhughes) [inline src/Settings_Helper.php]: Needs to be namespace aware.

**Review by barryhughes — CHANGES_REQUESTED** [review]
Thanks for the work on this!

One thing I'm less sure about is the inclusion of the Settings/Settings Helper classes. I think this is probably something where we ought to work changes back into Tribe Common. Could be worth talking that out in some more detail.

File: src/Settings.php (hunk 2 above)
Comment (cliffordp) [reply src/Settings.php]: done

File: src/Settings.php (hunk 3 above)
Comment (cliffordp) [reply src/Settings.php]: 👍

File: src/Settings_Helper.php (hunk 4 above)
Comment (cliffordp) [reply src/Settings_Helper.php]: Good catch. Added to main extension file's check as well.

**Comment by cliffordp** [comment]
@barryhughes 
I added some additional enhancements, particularly `README.md`
Please re-review.
Goal is to help quick-start newbies (our own or community) and to be able to remove the https://github.com/mt-support/extension-development-guide repo so we can maintain the template in a single location.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+1. Find and replace these strings throughout all file names and file contents:
+    1. Extension plugin's slug: `tribe-ext-extension-template`
+    1. Extension plugin's slug (yes, again): `match-the-plugin-directory-name`
+    1. `[Base Plugin Name] Extension: [Extension Name]`:
+        1. `[Base Plugin Name]` should be something like `The Events Calendar`
+        1. `[Extension Name]` should be the name of your extension, such as `Remove Export Links`
+    1. `[Extension Description]` with your extension plugin's short description.
+    1. PHP namespace: `Tribe\Extensions\Example`
+    1. Extension class (must be unique among all extensions): `Tribe__Extension__Example`
+    1.
```
Comment (barryhughes) [inline README.md]: Delete

File: README.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+1. Place the remainder of this template's files in your repo's new branch.
+1. Find and replace these strings throughout all file names and file contents:
+    1. Extension plugin's slug: `tribe-ext-extension-template`
+    1. Extension plugin's slug (yes, again): `match-the-plugin-directory-name`
+    1. `[Base Plugin Name] Extension: [Extension Name]`:
+        1. `[Base Plugin Name]` should be something like `The Events Calendar`
+        1. `[Extension Name]` should be the name of your extension, such as `Remove Export Links`
+    1. `[Extension Description]` with your extension plugin's short description.
+    1. PHP namespace: `Tribe\Extensions\Example`
+    1. Extension class (must be unique among all extensions): `Tribe__Extension__Example`
```
Comment (barryhughes) [inline README.md]: If we're encouraging PHP namespaces I'm not sure we also need double underscore pseudo-namespacing for the class name. Is this a limitation of the existing extension framework, or can we further simplify?

File: README.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+1. Remove the starter bits that don't apply to your extension, such as required PHP version comments.
+    1. Look for `TODO` comments to identify some things to modify or remove, as applicable. For example, remove the entire `src` directory and the `Tribe__Autoloader` code if you do not have any settings to add to the admin UI and do not need the autoloader for any other purpose.
+    1. It might be advisable to leave the PHP version checking logic (and matching part of readme.txt) unmodified until one of your last commits to ensure you correctly determine the required version.
+1. Add your filters and actions in the `init()` method, creating your custom methods as needed.
+1. Remove all unused/unnecessary code and comments, as this template may handle edge cases that do not apply to your extension.
+1. Sanitize all input and escape all output, as appropriate.
+1. Double-check your plugin's slug, name, and descriptions still apply to the end result of what your code actually accomplishes.
+1. Set the `== Changelog ==` section of `readme.txt`.
+1. Delete this `README.md` file.
+1. Zip and release your plugin!
```
Comment (barryhughes) [inline README.md:45]: I love that we started with a step to _"Create a new branch called `initial` or `v1` or whatever is appropriate"_ as it helps us set things up in a way that will make pull requests/initial code review easier, but it seems like we get to _"Zip and release your plugin!"_ without an actual code review step.

Especially for new extension authors, that's important — can we add it?

File: src/Settings_Helper.php [hunk 8]
```diff
@@ -4,7 +4,7 @@
 
 use Tribe__Main;
 
-if ( ! class_exists( 'Settings_Helper' ) ) {
+if ( ! class_exists( NS . 'Settings_Helper' ) ) {
```
Comment (barryhughes) [inline src/Settings_Helper.php]: :thinking: I'm not opposed to keeping these, but in a world where we encourage and give an example of autoloading, they are pretty unnecessary.

File: src/Settings.php (hunk 1 above)
Comment (cliffordp) [reply src/Settings.php]: missed this comment before. done.

File: README.md (hunk 5 above)
Comment (cliffordp) [reply README.md]: good catch on the empty line

File: README.md (hunk 6 above)
Comment (cliffordp) [reply README.md]: As discussed via Slack, went the route of custom namespace with `Main` class name

File: README.md (hunk 7 above)
Comment (cliffordp) [reply README.md:45]: As discussed via Slack, this README.md may be useful internally but is primarily for community usage.

File: src/Settings_Helper.php (hunk 8 above)
Comment (cliffordp) [reply src/Settings_Helper.php]: gotcha

**Review by barryhughes — APPROVED** [review]

**Pull request merged**